### PR TITLE
cortex: update alert and rule definitions

### PR DESCRIPTION
**Pull request by triclambert**
We're a bit behind https://github.com/grafana/cortex-jsonnet.

### CODE REVIEW

**Comment by triclambert** [comment]
WIP

```
0a1,2
> CortexAllocatingTooMuchMemory
> CortexBadRuntimeConfig
2,3c4,8
< CortexCacheRequestErrors
< CortexFlushStuck
---
> CortexCheckpointCreationFailed
> CortexCheckpointDeletionFailed
> CortexCompactorHasNotSuccessfullyCleanedUpBlocks
> CortexCompactorHasNotSuccessfullyRunCompaction
> CortexCompactorHasNotUploadedBlocks
5,6c10,11
< CortexGWRequestErrors
< CortexGWRequestLatency
---
> CortexGossipMembersMismatch
> CortexInconsistentRuntimeConfig
9a15,16
> CortexIngesterReachingSeriesLimit
> CortexIngesterReachingTenantsLimit
18a26,29
> CortexMemcachedRequestErrors
> CortexMemoryMapAreasTooHigh
> CortexOldChunkInMemory
> CortexProvisioningMemcachedTooSmall
21,22d31
< CortexProvisioningTooMuchMemory
< CortexQuerierCapacityFull
26d34
< CortexReadErrorBudgetBurn
29c37
< CortexRulerFailedEvaluations
---
> CortexRulerFailedRingCheck
30a39,41
> CortexRulerTooManyFailedPushes
> CortexRulerTooManyFailedQueries
> CortexSchedulerQueriesStuck
31a43
> CortexTableSyncFailure
34c46,47
< CortexWriteErrorBudgetBurn
---
> CortexWALCorruption
```

**Comment by triclambert** [comment]
Re-purposing PR for solely updating to cortex-mixin.  Force pushing to clean up history.

**Comment by MatApple** [comment]
Thanks @triclambert 🎉

**Comment by triclambert** [comment]
Testing this manually I see the (correct) addition of the new alerts (e.g., CortexIngesterHasUnshippedBlocks), but I still see some old ones (e.g., [Legacy*](https://system.tracy.opstrace.io/grafana/alerting/list?alertState=inactive&queryString=legacy)).

**Comment by triclambert** [comment]
Re-generating the alerts and rules from source:

```
gh repo clone grafana/cortex-jsonnet
cd cortex-jsonnet
make build-mixin
yq eval cortex-mixin/out/alerts.yaml -j -P > alerts.json
sed -i '' 's/#.*//' -e '^$' cortex-mixin/out/rules.yaml
yq eval cortex-mixin/out/rules.yaml -j -P > rules.json
```

Open the JSON in VScode and convert to a Javascript object.

Paste into `alerts.ts` and `rules.ts`.

**Comment by triclambert** [comment]
I have a cluster up with this change for manual inspection.  One of the rules seems to have a problem [1] and the Grafana 8 search doesn't seem to work [2].  But otherwise, things look ok.  

[1]
```
❯ kubectl logs -n cortex ruler-5f67cd6ccb-2mhqk | grep "parse error"   
level=error ts=2021-08-10T21:14:43.458722325Z caller=api.go:471 org_id=system traceID=1a4bf470296e4462 msg="unable to validate rule group payload" err="197:11: group \"cortex_scaling_rules\", rule 9, \"cluster_namespace_deployment:kube_pod_container_resource_requests_cpu_cores:sum\": could not parse expression: 1:9: parse error: unexpected \",\""
```

[2] https://system.mini3.opstrace.io/grafana/alerting/list?queryString=watchdog&dataSource=metrics

**Comment by sreis** [comment]
> the Grafana 8 search doesn't seem to work [2]

Looks like one endpoint is not found and websocket connection is always restarting.

Screenshot from 2021-08-11 13-46-42

File: packages/controller/src/system/rules.ts [hunk 1]
```diff
@@ -274,838 +274,510 @@ export default () => [
...
         record:
-          "namespace_job:cortex_gw_read_slo_errors_per_request:ratio_rate30m"
+          "cluster_namespace_deployment:container_cpu_usage_seconds_total:sum_rate"
       },
       {
-        expr: '1 -\n(\n  sum by (namespace, job) (rate(cortex_gw_request_duration_seconds_bucket{status_code!~"error|5..",le="2.5",route="cortex-read"}[1h]))\n/\n  sum by (namespace, job) (rate(cortex_gw_request_duration_seconds_count{route="cortex-read"}[1h]))\n)\n',
+        expr: '(cluster, namespace, deployment) (\n    label_replace(\n      label_replace(\n        kube_pod_container_resource_requests_cpu_cores,\n        "deployment", "$1", "pod", "(.*)-(?:([0-9]+)|([a-z0-9]+)-([a-z0-9]+))"\n      ),\n      \n      \n      "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"\n    )\n  )\n)\nor\n\n\n(\n  sum by (cluster, namespace, deployment) (\n    label_replace(\n      label_replace(\n        kube_pod_container_resource_requests{resource="cpu"},\n        "deployment", "$1", "pod", "(.*)-(?:([0-9]+)|([a-z0-9]+)-([a-z0-9]+))"\n      ),\n      \n      \n      "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"\n    )\n  )\n)\n',
```
Comment (sreis) [inline packages/controller/src/system/rules.ts:751]:
```suggestion
        expr: '(sum by (cluster, namespace, deployment) (\n    label_replace(\n      label_replace(\n        kube_pod_container_resource_requests_cpu_cores,\n        "deployment", "$1", "pod", "(.*)-(?:([0-9]+)|([a-z0-9]+)-([a-z0-9]+))"\n      ),\n      \n      \n      "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"\n    )\n  )\n)\nor\n\n\n(\n  sum by (cluster, namespace, deployment) (\n    label_replace(\n      label_replace(\n        kube_pod_container_resource_requests{resource="cpu"},\n        "deployment", "$1", "pod", "(.*)-(?:([0-9]+)|([a-z0-9]+)-([a-z0-9]+))"\n      ),\n      \n      \n      "deployment", "$1", "deployment", "(.*?)(?:-zone-[a-z])?"\n    )\n  )\n)\n',
```

Comment (triclambert) [reply packages/controller/src/system/rules.ts]: Pushed this fix.

I'm not sure how this happened.  I've checked from the [source](https://github.com/grafana/cortex-jsonnet/blob/dccf32abaa4af443d2b724edc80a7a28ec504c24/cortex-mixin/recording_rules.libsonnet#L229) through my process and can only conclude it was human error in the last possible stage.

**Comment by triclambert** [comment]
Manually checking on this, I think we're good to go:

```
$ kubectl logs -n cortex ruler-7458d44bb8-4h27r | grep "parse error"

$ kubectl logs -n cortex ruler-7458d44bb8-ndkpf | grep "parse error"

$ kubectl logs -n cortex ruler-7458d44bb8-zf2zw | grep "parse error"

```

**Comment by triclambert** [comment]
The upgrade test failed as of https://github.com/opstrace/opstrace/issues/293#issuecomment-896987225.  Merging.

**Pull request merged**